src/cli_common.c

```diff
@@ -427,25 +430,29 @@ sds cliVersion(void) {
 }
 
 /* This is a wrapper to call valkeyConnect or valkeyConnectWithTimeout. */
-valkeyContext *valkeyConnectWrapper(const char *ip, int port, const struct timeval tv, int nonblock) {
+valkeyContext *valkeyConnectWrapper(enum valkeyConnectionType ct, const char *ip_or_path, int port, const struct timeval tv, int nonblock) {
     valkeyOptions options = {0};
-    VALKEY_OPTIONS_SET_TCP(&options, ip, port);
 
-    if (tv.tv_sec || tv.tv_usec) {
-        options.connect_timeout = &tv;
-    }
+    switch (ct) {
+    case VALKEY_CONN_TCP:
+        VALKEY_OPTIONS_SET_TCP(&options, ip_or_path, port);
+        break;
 
-    if (nonblock) {
-        options.options |= VALKEY_OPT_NONBLOCK;
-    }
+    case VALKEY_CONN_UNIX:
+        VALKEY_OPTIONS_SET_UNIX(&options, ip_or_path);
+        break;
 
-    return valkeyConnectWithOptions(&options);
-}
+    case VALKEY_CONN_RDMA:
+#ifdef USE_RDMA
+        VALKEY_OPTIONS_SET_RDMA(&options, ip_or_path, port);
+        break;
+#else
+        assert(0); /* requesting RDMA connection without RDMA support??? */
```

Comment:
nit: can we exit with error? for precompiled valkey-cli user might have problems understanding the root cause.

Reply:
It can't happen. When compiling without RDMA, there is no --rdma option so the user can't get here. The assert is just for us and our CI to detect it and we want a stacktrace if it happens.

Reply:
Oh. I missed that :) O.K so I guess the comment can better explain it, but not a blocker.